server/routes/api/__test__/auth.unit.test.js

```diff
@@ -0,0 +1,48 @@
+const request = require('supertest');
+const app = require('../../../../app');
+const User = require('../../../models/user');
+
+describe('POST Auth/Login', () => {
+  beforeAll(() => {
+    return User.create({ email: 'testUser@test.com', password: 'test' });
+  });
+
+  afterAll(() => {
+    return User.findOneAndDelete({ email: 'testUser@test.com' });
+  });
+
+  it('will return error with invalid email', async () => {
+    const res = await request(app)
+      .post('/api/auth/login')
+      .type('form')
+      .send('email=test@test'); // send post with invalid invalid email
+    expect(res.statusCode).toEqual(400);
+    expect(res.body.error.message).toBe(
+      'Incorrect email and password combination.'
+    );
+  });
+
+  it('will return error with invalid password', async () => {
+    const res = await request(app)
+      .post('/api/auth/login')
+      .type('form')
+      .send('email=test@test.com')
```

Comment:
According to your description on 25, it seems like this test is meant to see if inputting the wrong password for an existing user will stop them from logging in. Because of that, it might make more sense to change the user email to be the same as the user we created before.

This is because we generally want to write each test to look at only one thing, that way if something does end up breaking it's easier to work backward and figure out what's causing it to break.

Reply:
Created a emailForTesting variable and am not using for all valid email posts.